src/blockdata/transaction.rs

```diff
@@ -378,9 +378,12 @@ impl Transaction {
             _ => unreachable!()
         };
         // hash the result
-        let mut raw_vec = serialize(&tx);
-        raw_vec.extend_from_slice(&endian::u32_to_array_le(sighash_u32));
-        SigHash::hash(&raw_vec)
+        // TODO: Sanity assert that consensus_encode returned length matches the hashed length in the hasher.
+        let sighash_arr = endian::u32_to_array_le(sighash_u32);
```

Comment:
Could you move this line down right above to where the variable is used?

Reply:
Done